extensions/wikia/UserLogin/FacebookSignupController.class.php

```diff
@@ -53,6 +53,12 @@ public function index() {
 				$user->setCookies();
 				$this->loggedIn = true;
 				$this->userName = $user->getName();
+
+				// Retrieve user email from Facebook if missing
+				$email = $user->getEmail();
+				if ( empty( $email ) ) {
+					$this->saveEmailAsynchronously( $user->getId() );
```

Comment:
My concern with this logic is that if a user signed up for facebook with a phone number and we don't have an email for them on our end, every time they log in with facebook on our site we'll kick off this saveEmailAsynchronously job which will always fail. Granted that should represent a small portion of our facebook users, but it's a condition which never resolves itself.

Maybe we could check if facebook has an email for them before kicking off the job? That does have the added weight of an additional request during login, but it would only be for that smaller subset of users and it would reduce the number of unnecessary jobs we create. Thoughts?